lib/model.js

```diff
@@ -1634,8 +1633,16 @@ function _ensureIndexes(model, options, callback) {
 
     const indexFields = utils.clone(index[0]);
     const indexOptions = utils.clone(index[1]);
+    const textBlock = () => {
+      const results = Object.keys(indexFields).map(function(key) {
+        return indexFields[key];
+      });
+      if (results.indexOf('text') !== -1) {
```

Comment:
Why not just check `indexFields.hasOwnProperty('text')`? That way you don't need this extra function, and it seems silly to convert an object to an array of keys just to use [`indexOf()`](https://masteringjs.io/tutorials/fundamentals/includes#indexof) to search through that array because `indexOf()` is much slower than accessing a document's keys.

More on [`hasOwnProperty()`](https://masteringjs.io/tutorials/fundamentals/hasownproperty).

Reply:
but doesn't `hasOwnProperty()` check the keys, not the value? When I run `indexFields.hasOwnProperty('text')` it returns false even though `'text'` does exist in `indexFields`.